event type use enum class

higumachan opened the pull request:
I use python enum class on "event_type". (https://docs.python.org/3.5/library/enum.html)

good points
- Easy to browse the event list
- If you use Pycharm or jedi or etc complementation tool then it will complement

alexchamberlain commented:
Thanks for the contribution @higumachan. My main thought when reviewing this is "Is this backwards compatible for existing users?"

Besides that, please review our [CONTRIBUTING.md][1]; it requires signing of commits to acknowledge the DCO et al. Please could you squash your commits into a single commit too?

[1]: https://github.com/bloomberg/python-github-webhook/blob/master/CONTRIBUTING.md

higumachan commented:
Thanks for reply.

This is not compatible for exiting users.
I will fix it to compatible way.

I reviewed "CONTRIBUTING.md".
I will squash my commits and add DCO.

higumachan committed:
Event type use enum class

I use python enum class on "event_type". (https://docs.python.org/3.5/library/enum.html)

good points

Easy to browse the event list
If you use Pycharm or jedi or etc complementation tool then it will complement

There feature is downward compatibility.

Signed-Off-By: Yuta Hinokuma <harekumo792154@gmail.com>

higumachan commented:
I fixed them.

Please could you review again?

edmorley commented on setup.py:
```diff
@@ -8,7 +8,7 @@
       license='Apache 2.0',
       packages=["github_webhook"],
-      install_requires=['flask', 'six'],
+      install_requires=['flask', 'six', 'enum34'],
```
Perhaps use the `python_version` specifier here to limit this dependency to Python 2 only?

edmorley commented on github_webhook/event_type.py:
```diff
@@ -0,0 +1,30 @@
+
+class EventType(Enum):
+    """
+    Event enum type
```
I think it would be good to link to https://developer.github.com/v3/activity/events/types/ in the docstring so it shows up in the docs, plus anyone keeping this event list up to date in the future knows where to look

edmorley commented on github_webhook/event_type.py:
```diff
@@ -0,0 +1,30 @@
+    Repository = 'repository'
+    Status = 'status'
+    TeamAdd = 'team_add'
+    Watch = 'watch'
```
There are a half dozen events that are missing here compared to the list on:
https://developer.github.com/v3/activity/events/types/

edmorley reviewed (commented):
I wonder whether this change actually makes things more confusing for users, given there's now another indirection between what they see on the GitHub docs and here?

It seems there are two things that could be improved compared to what's on master:
1) Making it easier for people to discover what hook names exist and what they do
2) Making it harder for people to typo those names

For (1), if people aren't using an IDE that recognises this enum, then I think this PR actually makes things worse, since now they can't just look at the GitHub docs, they have to look at the source for this package. (Yes this PR backward compat supports the string format, but users probably won't know that)

For (2), instead of using enums, the package could instead just validate the names passed against a fixed list that matches the GitHub docs names. This would obviously have to be kept up to date, but no more so than this enum. An added bonus of this approach would be that it's not a backward incompatible change either.
